src/OrleansAzureUtils/MultiClusterNetwork/AzureTableBasedGossipChannel.cs

```diff
@@ -0,0 +1,193 @@
+                    }
+                    catch (Exception exc)
+                    {
+                        logger.Error(
+                            ErrorCode.AzureTable_61,
+                            string.Format("Intermediate error parsing GossipTableEntry: {0}. Ignoring this entry.", tableEntry),
+                            exc);
+                    }
+                }
+            }
+
+            var writeback = new List<Task>();
+            var sendback = new Dictionary<SiloAddress,GatewayEntry>();
+
+            if (configToPush != null &&
+                (configRow == null || configRow.GossipTimestamp < configToPush.AdminTimestamp))
+            {
+                // push configuration
+                if (configRow == null)
+                    writeback.Add(tableManager.TryCreateConfigurationEntryAsync(configToPush));
+                else
+                    writeback.Add(tableManager.TryUpdateConfigurationEntryAsync(configToPush, configRow, configRow.ETag));
+            }
+            else if (configRow != null &&
+                 (configToPush == null || configToPush.AdminTimestamp < configRow.GossipTimestamp))
+            {
+                // pull configuration
+                returnedConfiguration = configRow.ToConfiguration();
+            }
+
+            foreach (var gatewayEntryPair in gateways.Values)
+            {
+                GatewayEntry gatewayEntry = gatewayEntryPair.Key;
+                GossipTableEntry tableEntry = gatewayEntryPair.Value;
+
+                if ((gatewayEntry != null && !gatewayEntry.Expired)
+                     && (tableEntry == null || tableEntry.GossipTimestamp < gatewayEntry.HeartbeatTimestamp))
+                {
+                    // push gateway entry, since we have a newer value
+                    if (tableEntry == null)
+                    {
+                        writeback.Add(tableManager.TryCreateGatewayEntryAsync(gatewayEntry));
+                    }
+                    else
+                    {
+                        writeback.Add(tableManager.TryUpdateGatewayEntryAsync(gatewayEntry, tableEntry, tableEntry.ETag));
+                    }
+                }
+                else if (tableEntry != null &&
+                        (gatewayEntry == null || gatewayEntry.HeartbeatTimestamp < tableEntry.GossipTimestamp))
+                {
+                    // pull or remove gateway entry
+                    gatewayEntry = tableEntry.ToGatewayEntry();
+                    if (gatewayEntry.Expired)
+                    {
+                        writeback.Add(tableManager.TryDeleteGatewayEntryAsync(tableEntry, tableEntry.ETag));
+                    }
+                    else
+                    {
+                        // gets sent back
+                        sendback.Add(tableEntry.SiloAddress, tableEntry.ToGatewayEntry());
+                    }
+                }
+            }
+
+            await Task.WhenAll(writeback);
```

Comment:
Since the "Push" method is used for pushing small deltas only (partial gossip, in contrast to the full gossip which calls PushAndPull), this code almost always reads/updates a single row only  (either the configuration row, or the row for the status of this gateway), and maybe two in very rare cases. Therefore I think it is not worth the extra code complexity to do batching.